File: app/src/main/java/com/lcl6/cn/basedialog/mvp/model/ManagerModel.java
package com.lcl6.cn.basedialog.mvp.model;

/**
 * Created by liancl on 2017/9/1.
 */

public interface ManagerModel {
    void requestTabaoData(LoadCompleteListener listener);
    void setGlobal(LoadCompleteListener listener,String newUrl);
    void removeGlobal(LoadCompleteListener listener);
    void requestGuthub(LoadCompleteListener listener,String newUrl);
    void requestGank(LoadCompleteListener listener,String newUrl);
    void requestDouban(LoadCompleteListener listener,String newUrl);

    interface LoadCompleteListener{
        void onComplete(String successMessage);
        void onFail(String errorMessage);
    }

}

File: app/src/main/java/com/lcl6/cn/basedialog/mvp/model/ManagerModel.kt
package com.lcl6.cn.basedialog.mvp.model

/**
 * Created by liancl on 2017/9/1.
 */

interface ManagerModel {
    fun requestTabaoData(listener: LoadCompleteListener)
    fun setGlobal(listener: LoadCompleteListener, newUrl: String)
    fun removeGlobal(listener: LoadCompleteListener)
    fun requestGuthub(listener: LoadCompleteListener, newUrl: String)
    fun requestGank(listener: LoadCompleteListener, newUrl: String)
    fun requestDouban(listener: LoadCompleteListener, newUrl: String)

    interface LoadCompleteListener {
        fun onComplete(successMessage: String)
        fun onFail(errorMessage: String)
    }

}
